components/scheduler/src/Scheduler.svelte

```diff
@@ -1,24 +1,94 @@
 <svelte:options tag="nylas-scheduler" />
 
 <script lang="ts">
-  import { ManifestStore } from "../../../commons/src";
+  import {
+    ManifestStore,
+    EventStore,
+    AvailabilityStore,
+  } from "../../../commons/src";
+  import { createEvent } from "@commons/connections/events";
+
   import { onMount, tick } from "svelte";
   import "../../availability/src/Availability.svelte";
 
-  export let id: string = "";
-  export let access_token: string = "";
+  import {
+    getPropertyValue,
+    buildInternalProps,
+  } from "@commons/methods/component";
 
   import type { Manifest } from "@commons/types/Scheduler";
+  import type { TimeSlot } from "@commons/types/Availability";
+  import type { TimespanEvent, EventQuery } from "@commons/types/Events";
 
+  // #region props
+  export let id: string = "";
+  export let access_token: string = "";
+  export let availability_id: string;
+  export let email_ids: string[];
+  // #endregion props
+
+  //#region mount and prop initialization
+  let internalProps: Partial<Manifest> = {};
   let manifest: Partial<Manifest> = {};
   onMount(async () => {
     await tick();
     const storeKey = JSON.stringify({ component_id: id, access_token });
     manifest = (await $ManifestStore[storeKey]) || {};
+
+    internalProps = buildInternalProps($$props, manifest) as Partial<Manifest>;
   });
 
-  function timeSlotChosen({ detail }) {
-    console.log("timeslots chosen", detail.timeSlots);
+  $: {
+    const rebuiltProps = buildInternalProps(
+      $$props,
+      manifest,
+    ) as Partial<Manifest>;
+    if (JSON.stringify(rebuiltProps) !== JSON.stringify(internalProps)) {
+      internalProps = rebuiltProps;
+    }
+  }
```

Comment:
I'm a little confused by this block. It looks like `internalProps` is set, then when `$$props` or `manifest` changes, `rebuiltProps` is set. Why not stick with `internalProps` as a reactive variable?

Reply:
@ozsivanov might be the best person to speak to this

Reply:
@heathergh, it has to be done this way to prevent the `internalProps` reactive binding from firing when there have been no changes made to it. This was causing all of the `getPropertyValue()` bindings to fire, which in turn caused dependent bindings to fire as well. 

This caused an issue in the availability component in particular, where clicking on a timeslot would cause the `internalProps` binding to fire, rebuilding internal props which would fire the reactive binding for all of its props, including `start_time`. `start_time` was used in a reactive binding that build all of the timeslots, which would cause the clicked timeSlot to be unselected immediately.